### PR TITLE
Fix support for write-only properties in BeEquivalentTo

### LINKED ISSUE #2832: Interface with Setter Only Property leads to NullReferenceExcpetion
### Description

When asserting an object with `BeEquivalentTo` and the type of that object implements an interface with a setter only property, a `NullReferenceException` is thrown

### Reproduction Steps

```csharp
[TestClass]
public class Test
{
    [TestMethod]
    public void Issue()
    {
        var a = new AClass { Value = "OtherVlaue" };
        var b = new AClass { Value = "OtherVlaue" };
        
        a.Should().BeEquivalentTo(b);
    }
    
    private class AClass : IWithSetterOnlyProperty
    {
        public string Value { private get; set; }
    }
    
    private interface IWithSetterOnlyProperty
    {
        string Value { set; }
    }
}
```

### Expected behavior

No `NullReferenceException` should be thrown.

### Actual behavior

A `NullReferenceException` is thrown by Reflectify. It looks like the issue was introduce with https://github.com/fluentassertions/fluentassertions/pull/2794

### Regression?

_No response_

### Known Workarounds

Adding a getter to the property solves the issue.

### Configuration

_No response_

### Other information

Call stack of the `NullReferenceException`:
```csharp
System.NullReferenceException: Object reference not set to an instance of an object.
   at Reflectify.Reflector.AddInterfaceProperties(Type typeToReflect, MemberKind kind)
   at Reflectify.Reflector.LoadProperties(Type typeToReflect, MemberKind kind)
   at Reflectify.Reflector..ctor(Type typeToReflect, MemberKind kind)
   at Reflectify.TypeExtensions.<>c.<GetFor>b__4_0(ValueTuple`2 key)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Reflectify.TypeExtensions.GetFor(Type typeToReflect, MemberKind kind)
   at Reflectify.TypeExtensions.GetProperties(Type type, MemberKind kind)
   at FluentAssertions.Equivalency.Selection.AllPropertiesSelectionRule.SelectMembers(INode currentNode, IEnumerable`1 selectedMembers, MemberSelectionContext context)
   at FluentAssertions.Equivalency.Steps.StructuralEqualityEquivalencyStep.GetMembersFromExpectation(INode currentNode, Comparands comparands, IEquivalencyAssertionOptions options)
   at FluentAssertions.Equivalency.Steps.StructuralEqualityEquivalencyStep.Handle(Comparands comparands, IEquivalencyValidationContext context, IEquivalencyValidator nestedValidator)
   at FluentAssertions.Equivalency.EquivalencyValidator.TryToProveNodesAreEquivalent(Comparands comparands, IEquivalencyValidationContext context)
   at FluentAssertions.Equivalency.EquivalencyValidator.RecursivelyAssertEquality(Comparands comparands, IEquivalencyValidationContext context)
   at FluentAssertions.Equivalency.EquivalencyValidator.AssertEquality(Comparands comparands, EquivalencyValidationContext context)
   at FluentAssertions.Primitives.ObjectAssertions`2.BeEquivalentTo[TExpectation](TExpectation expectation, Func`2 config, String because, Object[] becauseArgs)
   at FluentAssertions.Primitives.ObjectAssertions`2.BeEquivalentTo[TExpectation](TExpectation expectation, String because, Object[] becauseArgs)

```
### Are you willing to help with a pull-request?

No

### PR DESCRIPTION
**Pull request by dennisdoomen**
Fixes #2832 

The implementation of `Type.GetProperties` that [Reflectify](https://github.com/dennisdoomen/reflectify) provided crashed on a write-only property that is part of an interface. We didn't have a test for that, and the only other test that covered a similar scenario wasn't properly implemented.

**Note** Needs to rebased when #2835 is merged. 

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Comment by jnyrup** [comment]
I'll try to have a look today.

**Comment by IT-VBFK** [comment]
Is it intentional that this PR also contains the "Drop support..." commit?

**Comment by dennisdoomen** [comment]
> Is it intentional that this PR also contains the "Drop support..." commit?

Yes, it needs to rebased when #2835 is merged.

**Review by jnyrup — COMMENTED** [review]
Found a small typo.

I tried reproducing the reported bug in #2832 locally to see what caused the problems, but didn't finished. Will continue later tonight

**Review by jnyrup — APPROVED** [review]

File: docs/_pages/releases.md [hunk 1]
```diff
@@ -14,6 +14,10 @@ sidebar:
 * Dropped direct support for .NET Core 2.x and .NET Core 3.x - [#2302](https://github.com/fluentassertions/fluentassertions/pull/2302)
 * Dropped support for `NSpec3` test framework - [#2356](https://github.com/fluentassertions/fluentassertions/pull/2356)
 
+### Fixes
+
+* Fixed a problem in BeEquivalentTo where write-only properties would cause a NRE - [#2836](https://github.com/fluentassertions/fluentassertions/pull/2836)
```
[the remark this answers is not in the document]
Comment (dennisdoomen) [reply docs/_pages/releases.md:19]: > "a NRE" -> "an NRE"

It depends if you are silenty saying "an N R E" or "a NullReferenceException" -)

**Pull request merged**